packages/dapp-kit-next/src/core/store.ts

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import type { UiWallet, UiWalletAccount } from '@wallet-standard/ui';
+import { atom, computed, map } from 'nanostores';
+import { getChain } from '../utils/networks.js';
+import type { Networks } from '../utils/networks.js';
+import { getAssociatedWalletOrThrow, requiredWalletFeatures } from '../utils/wallets.js';
+import { publicKeyFromSuiBytes } from '@mysten/sui/verify';
+import type { Experimental_BaseClient } from '@mysten/sui/experimental';
+
+type WalletConnection =
+	| {
+			status: 'disconnected' | 'connecting';
+			currentAccount: null;
+	  }
+	| {
+			status: 'reconnecting' | 'connected';
+			currentAccount: UiWalletAccount;
+	  };
+
+export type DAppKitStores<TNetworks extends Networks = Networks> = ReturnType<
+	typeof createStores<TNetworks>
+>;
+
+export function createStores<TNetworks extends Networks>({
+	defaultNetwork,
+	getClient,
+}: {
+	defaultNetwork: TNetworks[number];
+	getClient: (network: TNetworks[number]) => Experimental_BaseClient;
+}) {
+	const $baseConnection = map<WalletConnection>({
```

Comment:
The only downside to having multiple stores is that it's a _little bit_ easier to have state drift if something unexpected happens (method throws). In practice this just means we need to think about these cases and test for them to ensure this doesn't happen, so not really a concern for this PR but just something to keep in mind since atom state changes can be disconnected now.